src/librbd/cache/rwl/Request.cc

```diff
@@ -0,0 +1,369 @@
+// -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*-
+// vim: ts=8 sw=2 smarttab
+
+#include "Request.h"
+#include "librbd/BlockGuard.h"
+#include "librbd/cache/rwl/LogEntry.h"
+
+#define dout_subsys ceph_subsys_rbd_rwl
+#undef dout_prefix
+#define dout_prefix *_dout << "librbd::cache::rwl::Request: " << this << " " \
+                           <<  __func__ << ": "
+
+namespace librbd {
+namespace cache {
+namespace rwl {
+
+typedef std::list<std::shared_ptr<GenericWriteLogEntry>> GenericWriteLogEntries;
+
+template <typename T>
+C_BlockIORequest<T>::C_BlockIORequest(T &rwl, const utime_t arrived, io::Extents &&extents,
+                                      bufferlist&& bl, const int fadvise_flags, Context *user_req)
+  : rwl(rwl), image_extents(std::move(extents)),
+    bl(std::move(bl)), fadvise_flags(fadvise_flags),
+    user_req(user_req), image_extents_summary(image_extents), m_arrived_time(arrived) {
+  ldout(rwl.get_context(), 99) << this << dendl;
+  /* Remove zero length image extents from input */
```

Comment:
Nit: this is handled at a higher level now, right?

Reply:
I added a change here https://github.com/ceph/ceph/blob/master/src/librbd/io/ImageRequest.cc#L424-L430 to check whether the total length is 0.  
I also think the handler should be at a higher level. But from the existed codes, I didn't find a case that there are more than 1 extents in an ImageRequest. Do you think we need to add the handler?

Reply:
I *believe* support is only there to handle reads from parent images -- the write path is unused at the moment.

Reply:
C_BlockIORequest is a parent class for write/flush/discard/writesame/compareandwrite requests. And it is not a parent class of read request. So this problem should not exist.